# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/QuicSocketControl.cpp
# Commit: 0bf045553a40
# Full Hash: 0bf045553a4079b9948e1814e5cd1038be301b0b
# Author: Dragana Damjanovic <dd.mozilla@gmail.com>
# Date: 2019-11-02 09:54:21
# Regressor Bug: 1581637
# File Overlap Count: 1
# Description:
#   Bug 1581637 - Part 8 - Add Http3Session/Http3Stream. r=mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D46652
# ==============================================================================

diff -r 2801613e5d2d -r 0bf045553a40 netwerk/protocol/http/QuicSocketControl.cpp
--- a/netwerk/protocol/http/QuicSocketControl.cpp	Fri Nov 01 14:55:56 2019 +0000
+++ b/netwerk/protocol/http/QuicSocketControl.cpp	Fri Nov 01 14:55:55 2019 +0000
@@ -43,20 +43,17 @@
 }
 
 void QuicSocketControl::CallAuthenticated() {
-  //  // Will be added when Http3 lands
-  /*  if (mHttp3Session) {
+  if (mHttp3Session) {
     RefPtr<Http3Session> http3Session = do_QueryReferent(mHttp3Session);
     http3Session->Authenticated(GetErrorCode());
   }
-  mHttp3Session = nullptr;*/
+  mHttp3Session = nullptr;
 }
 
-// Will be added when Http3 lands
-// void QuicSocketControl::SetAuthenticationCallback(Http3Session
-// *aHttp3Session) {
-//  mHttp3Session = do_GetWeakReference(
-//      static_cast<nsISupportsWeakReference*>(aHttp3Session));
-//}
+void QuicSocketControl::SetAuthenticationCallback(Http3Session* aHttp3Session) {
+  mHttp3Session = do_GetWeakReference(
+      static_cast<nsISupportsWeakReference*>(aHttp3Session));
+}
 
 void QuicSocketControl::HandshakeCompleted() {
   psm::RememberCertErrorsTable::GetInstance().LookupCertErrorBits(this);